src/DAppProviders.tsx

```diff
@@ -0,0 +1,14 @@
+import React from "react"
+import { ChakraProvider } from "@chakra-ui/react"
+import theme from "./theme"
+import DApp from "./DApp"
+
+function DAppProviders() {
```

Comment:
Have you considered including [such utility](https://javascript.plainenglish.io/how-to-combine-context-providers-for-cleaner-react-code-9ed24f20225e)? The approach you propose has few drawbacks:
- it will get difficult to read and maintain as applications scales:
```jsx
<ProviderOne>
  <ProviderTwo>
    <ProviderThree>
      <ProviderFour>
        {...} 
      </ProviderFour>
    </ProviderThree>
  </ProviderTwo>
</ProviderOne>
```
- in some cases the order of provider matters so some components might not have access to the data provided by specific provider

This simple utility resolves both problems. It can be written by hand - just a simple wrapper around `reduce` array method or installed as a package - there are a bunch of them available.